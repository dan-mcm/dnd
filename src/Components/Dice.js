import React, { Component } from "react";
import styled from "styled-components";
import { Button } from 'antd';
import DiceResult from './DiceResult';

const Centered = styled.div`
  width: 10%;
  margin: auto;
  background-color: darkred;
  color: white;
  border: 2px solid grey;
  padding: 20px;
  text-align: center;
  display: inline-block;
  margin: 20px;
`;

const Dimage = styled.img`
  width: 100px;
  height: 100px;
`;

const Score = styled.p`
  font-size: large;
`;

class Dice extends Component {
  state = {
    numberDice: 0,
    showResults: true,
    diceResults: []
  };

  getResults(event) {
    const numberOfDice = event.target.value;
    const diceResults = [];
    for (let i = 0; i < numberOfDice; i++) {
      const rollResult = Math.floor(Math.random() * this.props.sides + 1);
      diceResults.push(rollResult);
    }
    this.setState({
      numberDice: numberOfDice,
      diceResults
    });
  }

  roll(){
    const diceResults = [];
    for (let i = 0; i < this.state.numberDice; i++) {
      const rollResult = Math.floor(Math.random() * this.props.sides + 1);
      diceResults.push(rollResult);
    }

    this.setState({
      showResults: true,
      diceResults,
    })

    this.props.update(this.state.showResults, this.props.title, this.state.numberDice, this.state.diceResults);
  }

  render() {
    return (
      <Centered>
        <Dimage src={this.props.image} />
        <h2>{this.props.title}</h2>
        <select
          onChange={value => this.getResults(value)}
        >
          <option value="-">-</option>
          <option value="1">1</option>
          <option value="2">2</option>
          <option value="3">3</option>
          <option value="4">4</option>
          <option value="5">5</option>
          <option value="6">6</option>
          <option value="7">7</option>
          <option value="8">8</option>
          <option value="9">9</option>
          <option value="10">10</option>
        </select>
        <br />
        <br />

          <Button onClick={() => {this.roll()}}>Roll?</Button>

        <br />
        <br />

      </Centered>
    );
  }
}

export default Dice;
