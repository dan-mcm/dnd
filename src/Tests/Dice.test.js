import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Dice from '../Components/Dice';
import D4 from "../Images/D4.png";
import D6 from "../Images/D6.png";
import D8 from "../Images/D8.png";
import D10 from "../Images/D10.png";
import D12 from "../Images/D12.png";
import D20 from "../Images/D20.png";
import D100 from "../Images/D100.png";

//TODO -> inspect snapshot results

it('Renders a D4 Dice component snapshot', () => {
  const tree = renderer.create(<Dice image={D4} title={"D4"}/>);
  expect(tree.toJSON()).toMatchSnapshot();
});

it('Renders a D6 Dice component snapshot', () => {
  const tree = renderer.create(<Dice image={D6} title={"D6"}/>);
  expect(tree.toJSON()).toMatchSnapshot();
});

it('Renders a D8 Dice component snapshot', () => {
  const tree = renderer.create(<Dice image={D8} title={"D8"}/>);
  expect(tree.toJSON()).toMatchSnapshot();
});

it('Renders a D10 Dice component snapshot', () => {
  const tree = renderer.create(<Dice image={D10} title={"D10"}/>);
  expect(tree.toJSON()).toMatchSnapshot();
});

it('Renders a D12 Dice component snapshot', () => {
  const tree = renderer.create(<Dice image={D12} title={"D12"}/>);
  expect(tree.toJSON()).toMatchSnapshot();
});

it('Renders a D20 Dice component snapshot', () => {
  const tree = renderer.create(<Dice image={D20} title={"D20"}/>);
  expect(tree.toJSON()).toMatchSnapshot();
});

it('Renders a D100 Dice component snapshot', () => {
  const tree = renderer.create(<Dice image={D100} title={"D100"}/>);
  expect(tree.toJSON()).toMatchSnapshot();
});
