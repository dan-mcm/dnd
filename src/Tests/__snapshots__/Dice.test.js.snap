// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a D4 Dice component snapshot 1`] = `
<div
  className="sc-bwzfXH jlBZef"
>
  <img
    className="sc-htpNat hLAwSI"
    src="D4.png"
  />
  <h2>
    D4
  </h2>
  <select
    onChange={[Function]}
  >
    <option
      value="-"
    >
      -
    </option>
    <option
      value="1"
    >
      1
    </option>
    <option
      value="2"
    >
      2
    </option>
    <option
      value="3"
    >
      3
    </option>
    <option
      value="4"
    >
      4
    </option>
    <option
      value="5"
    >
      5
    </option>
    <option
      value="6"
    >
      6
    </option>
    <option
      value="7"
    >
      7
    </option>
    <option
      value="8"
    >
      8
    </option>
    <option
      value="9"
    >
      9
    </option>
    <option
      value="10"
    >
      10
    </option>
  </select>
  <br />
  <br />
  <button
    onClick={[Function]}
  >
    Roll?
  </button>
  <br />
  <br />
</div>
`;

exports[`Renders a D6 Dice component snapshot 1`] = `
<div
  className="sc-bwzfXH jlBZef"
>
  <img
    className="sc-htpNat hLAwSI"
    src="D6.png"
  />
  <h2>
    D6
  </h2>
  <select
    onChange={[Function]}
  >
    <option
      value="-"
    >
      -
    </option>
    <option
      value="1"
    >
      1
    </option>
    <option
      value="2"
    >
      2
    </option>
    <option
      value="3"
    >
      3
    </option>
    <option
      value="4"
    >
      4
    </option>
    <option
      value="5"
    >
      5
    </option>
    <option
      value="6"
    >
      6
    </option>
    <option
      value="7"
    >
      7
    </option>
    <option
      value="8"
    >
      8
    </option>
    <option
      value="9"
    >
      9
    </option>
    <option
      value="10"
    >
      10
    </option>
  </select>
  <br />
  <br />
  <button
    onClick={[Function]}
  >
    Roll?
  </button>
  <br />
  <br />
</div>
`;

exports[`Renders a D8 Dice component snapshot 1`] = `
<div
  className="sc-bwzfXH jlBZef"
>
  <img
    className="sc-htpNat hLAwSI"
    src="D8.png"
  />
  <h2>
    D8
  </h2>
  <select
    onChange={[Function]}
  >
    <option
      value="-"
    >
      -
    </option>
    <option
      value="1"
    >
      1
    </option>
    <option
      value="2"
    >
      2
    </option>
    <option
      value="3"
    >
      3
    </option>
    <option
      value="4"
    >
      4
    </option>
    <option
      value="5"
    >
      5
    </option>
    <option
      value="6"
    >
      6
    </option>
    <option
      value="7"
    >
      7
    </option>
    <option
      value="8"
    >
      8
    </option>
    <option
      value="9"
    >
      9
    </option>
    <option
      value="10"
    >
      10
    </option>
  </select>
  <br />
  <br />
  <button
    onClick={[Function]}
  >
    Roll?
  </button>
  <br />
  <br />
</div>
`;

exports[`Renders a D10 Dice component snapshot 1`] = `
<div
  className="sc-bwzfXH jlBZef"
>
  <img
    className="sc-htpNat hLAwSI"
    src="D10.png"
  />
  <h2>
    D10
  </h2>
  <select
    onChange={[Function]}
  >
    <option
      value="-"
    >
      -
    </option>
    <option
      value="1"
    >
      1
    </option>
    <option
      value="2"
    >
      2
    </option>
    <option
      value="3"
    >
      3
    </option>
    <option
      value="4"
    >
      4
    </option>
    <option
      value="5"
    >
      5
    </option>
    <option
      value="6"
    >
      6
    </option>
    <option
      value="7"
    >
      7
    </option>
    <option
      value="8"
    >
      8
    </option>
    <option
      value="9"
    >
      9
    </option>
    <option
      value="10"
    >
      10
    </option>
  </select>
  <br />
  <br />
  <button
    onClick={[Function]}
  >
    Roll?
  </button>
  <br />
  <br />
</div>
`;

exports[`Renders a D12 Dice component snapshot 1`] = `
<div
  className="sc-bwzfXH jlBZef"
>
  <img
    className="sc-htpNat hLAwSI"
    src="D12.png"
  />
  <h2>
    D12
  </h2>
  <select
    onChange={[Function]}
  >
    <option
      value="-"
    >
      -
    </option>
    <option
      value="1"
    >
      1
    </option>
    <option
      value="2"
    >
      2
    </option>
    <option
      value="3"
    >
      3
    </option>
    <option
      value="4"
    >
      4
    </option>
    <option
      value="5"
    >
      5
    </option>
    <option
      value="6"
    >
      6
    </option>
    <option
      value="7"
    >
      7
    </option>
    <option
      value="8"
    >
      8
    </option>
    <option
      value="9"
    >
      9
    </option>
    <option
      value="10"
    >
      10
    </option>
  </select>
  <br />
  <br />
  <button
    onClick={[Function]}
  >
    Roll?
  </button>
  <br />
  <br />
</div>
`;

exports[`Renders a D20 Dice component snapshot 1`] = `
<div
  className="sc-bwzfXH jlBZef"
>
  <img
    className="sc-htpNat hLAwSI"
    src="D20.png"
  />
  <h2>
    D20
  </h2>
  <select
    onChange={[Function]}
  >
    <option
      value="-"
    >
      -
    </option>
    <option
      value="1"
    >
      1
    </option>
    <option
      value="2"
    >
      2
    </option>
    <option
      value="3"
    >
      3
    </option>
    <option
      value="4"
    >
      4
    </option>
    <option
      value="5"
    >
      5
    </option>
    <option
      value="6"
    >
      6
    </option>
    <option
      value="7"
    >
      7
    </option>
    <option
      value="8"
    >
      8
    </option>
    <option
      value="9"
    >
      9
    </option>
    <option
      value="10"
    >
      10
    </option>
  </select>
  <br />
  <br />
  <button
    onClick={[Function]}
  >
    Roll?
  </button>
  <br />
  <br />
</div>
`;

exports[`Renders a D100 Dice component snapshot 1`] = `
<div
  className="sc-bwzfXH jlBZef"
>
  <img
    className="sc-htpNat hLAwSI"
    src="D100.png"
  />
  <h2>
    D100
  </h2>
  <select
    onChange={[Function]}
  >
    <option
      value="-"
    >
      -
    </option>
    <option
      value="1"
    >
      1
    </option>
    <option
      value="2"
    >
      2
    </option>
    <option
      value="3"
    >
      3
    </option>
    <option
      value="4"
    >
      4
    </option>
    <option
      value="5"
    >
      5
    </option>
    <option
      value="6"
    >
      6
    </option>
    <option
      value="7"
    >
      7
    </option>
    <option
      value="8"
    >
      8
    </option>
    <option
      value="9"
    >
      9
    </option>
    <option
      value="10"
    >
      10
    </option>
  </select>
  <br />
  <br />
  <button
    onClick={[Function]}
  >
    Roll?
  </button>
  <br />
  <br />
</div>
`;
